eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/DefaultEventStoreTransactionTest.java

```diff
@@ -141,6 +143,58 @@ void sourceReturnsOnlyCommitedEvents() {
                         .assertNext(entry -> assertTagsPositionAndEvent(entry, TEST_AGGREGATE_CRITERIA, 1, event2))
                         .verifyComplete();
         }
+
+        @Test
+        void appendCommitsOfNonExistentTagWhenOfTwoNonOverlappingTagsOneYieldedNoEvents() {
+
+            Tag nonExistentTag = new Tag("nonExistent", "tag");
+            EventCriteria nonExistingCriteria = EventCriteria.forAnyEventType().withTags(nonExistentTag);
+            Tag existentTag = new Tag("existent", "tag");
+            EventCriteria existingCriteria = EventCriteria.forAnyEventType().withTags(existentTag);
+
+            appendExistingEventForTag(existentTag);
+            testCanCommitTag(nonExistingCriteria, existingCriteria, nonExistentTag);
+        }
+
+        @Test
+        void appendCommitsOfExistentTagWhenOfTwoNonOverlappingTagsOneYieldedNoEvents() {
+
+            Tag nonExistentTag = new Tag("nonExistent", "tag");
+            EventCriteria nonExistingCriteria = EventCriteria.forAnyEventType().withTags(nonExistentTag);
+            Tag existentTag = new Tag("existent", "tag");
+            EventCriteria existingCriteria = EventCriteria.forAnyEventType().withTags(existentTag);
+
+            appendExistingEventForTag(existentTag);
+            testCanCommitTag(nonExistingCriteria, existingCriteria, existentTag);
+        }
+
+        private ConsistencyMarker appendExistingEventForTag(Tag existentTag) {
+            return eventStorageEngine.appendEvents(AppendCondition.none(),
+                                                   new GenericTaggedEventMessage<>(
+                                                           new GenericEventMessage<>(new MessageType(String.class),
+                                                                                     "my payload"),
+                                                           Set.of(existentTag)
+                                                   )).join().commit().join();
+        }
```

Comment:
Nit: In the context of this method, prepending the parameter name `existent` is not necessary.
```suggestion
        private ConsistencyMarker appendExistingEventForTag(Tag tag) {
            return eventStorageEngine.appendEvents(AppendCondition.none(),
                                                   new GenericTaggedEventMessage<>(
                                                           new GenericEventMessage<>(new MessageType(String.class),
                                                                                     "my payload"),
                                                           Set.of(tag)
                                                   )).join().commit().join();
        }
```